.github/workflows/update-3rdparty.yml

```diff
@@ -0,0 +1,254 @@
+          echo "$lines_changed lines changed."
+          echo ::set-output name=lines_changed::$lines_changed
+          
+      - name: send in a pull request
+        uses: gr2m/create-or-update-pull-request-action@v1.x
+        if: steps.getdiff.outputs.lines_changed > 0
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          title: "Update bundled IRE mapper script"
+          body: |
+            #### Brief overview of PR changes/additions
+            :crown: An automated PR to update the IRE mapper script to the latest version.
+            #### Motivation for adding to Mudlet
+            So people don't get attacked with a "your mapping script is out of date!" notification as soon as they create a profile for a new IRE game.
+            #### Other info (issues closed, discussion etc)
+            _update triggered by ${{ github.ref }} ${{ github.sha }}_
+          branch: update-ire-mapping-script-${{ github.sha }}
+          path: src/
+          commit-message: (autocommit) Updated IRE mapping script to latest upstream
+          author: "mudlet-machine-account <mudlet-machine-account@users.noreply.github.com>"
+
+  update-edbee-fork:
+    runs-on: ubuntu-latest
+    steps:      
+      - uses: actions/checkout@master
+        with:
+          ref: development
+          submodules: true
+          fetch-depth: 0
+
+      - name: update submodule to latest
+        run: git submodule update --remote 3rdparty/edbee-lib/
+        
+      # strictly speaking this step isn't necessary since the PR action can filter as well, but it's easier
+      # to see in github UI's if PR creation was skipped when we have an explicit pre-check
+      - name: check if we have any updates
+        id: getdiff
+        run: |
+          lines_changed=$(git diff --patch --unified=0 3rdparty/edbee-lib/ | wc --lines)
+          echo "$lines_changed lines changed."
+          echo ::set-output name=lines_changed::$lines_changed
+          
+      - name: send in a pull request
+        uses: gr2m/create-or-update-pull-request-action@v1.x
+        if: steps.getdiff.outputs.lines_changed > 0
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          title: "Update edbee to latest"
+          body: |
+            #### Brief overview of PR changes/additions
+            :crown: An automated PR to update edbee to its latest version in our fork.
+            #### Motivation for adding to Mudlet
+            Get new features, bugfixes, improvements! Stay modern.
+            #### Other info (issues closed, discussion etc)
+            _update triggered by ${{ github.ref }} ${{ github.sha }}_
+          branch: update-edbee-lib-${{ github.sha }}
+          path: 3rdparty/edbee-lib
+          commit-message: (autocommit) Updated edbee-lib submodule to latest in our fork
+          author: "mudlet-machine-account <mudlet-machine-account@users.noreply.github.com>"
+
+  update-sparkle-glue-fork:
```

Comment:
There's a fair bit of repetition in the PR. I'm not sure on whenever this is good or bad; whenever adding another layer of abstraction will help or not.

Reply:
I'd leave it as-is for now. If we see the need to keep modifying these in the same way, we can build the abstraction then.